src/linter/types.go

```diff
@@ -101,6 +102,13 @@ func (conv *phpdocTypeConverter) mapType(e phpdoc.TypeExpr) []meta.Type {
 			types = append(types, conv.mapType(a)...)
 		}
 		return types
+
+	case phpdoc.ExprOptional:
+		// If the type does not denote the shape or tuple type,
+		// then it is most likely an invalid nullable type syntax.
+		if !strings.Contains(e.Value, "shape") && !strings.Contains(e.Value, "tuple") {
```

Comment:
Tuples can't have optional keys.

Reply:
Also, I don't think `e.Value` would contain what you're looking for. You probably need to save the context like we do for the nullable types.

Reply:
ping